src/js/view/view.js

```diff
@@ -858,7 +858,19 @@ define([
 
         function _onMediaTypeChange(model, val) {
             var isAudioFile = (val ==='audio');
+            var provider = _model.getVideo();
+            var isNotFlash = (provider && provider.getName().name.indexOf('flash') !== 0);
+            var previewElem = _playerElement.getElementsByClassName('jw-preview')[0];
```

Comment:
We want to avoid DOM searches when possible. Can you think of a better way to do lines 863 and 864?